subsys/ieee802154/Kconfig

```diff
@@ -0,0 +1,204 @@
+config IEEE802154_NRF5_ANT_DIVERSITY_PIN
+	int "Antenna selection pin"
+	default 20
+	help
+	  Pin used for antenna selection.
+
+config IEEE802154_NRF5_ANT_DIVERSITY_TOGGLE_TIME
+	int "Toggle time"
+	default 13
+	help
+	  Time between antenna switches in automatic mode [us]
+endif
+
+config IEEE802154_NRF5_WIFI_COEX
+	bool "Enable WiFi Coexistance"
+	default n
+	help
+	  Enable WiFi Coexistance
+
+if IEEE802154_NRF5_WIFI_COEX
+
+choice IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE
+	prompt "WiFi Coex RX request mode"
+	default IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE_ED
+	help
+	  Mode of triggering receive request to Coex arbiter
+
+config IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE_ED
+	bool "Energy detected"
+
+config IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE_PREAMBLE
+	bool "Preamble reception"
+
+config IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE_DESTINED
+	bool "Frame detected"
+
+endchoice
+
+choice IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE
+	prompt "WiFi Coex TX request mode"
+	default IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE_FRAME_READY
+	help
+	  Mode of triggering transmit request to Coex arbiter
+
+config IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE_FRAME_READY
+	bool "Frame ready"
+
+config IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE_CCA_START
+	bool "Pre CCA"
+
+config IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE_CCA_DONE
+	bool "Post CCA"
+
+endchoice
+endif
+
+config IEEE802154_NRF5_FEM_PRESENT
+	bool "Enable Front End Module (FEM) control"
+	default n
+	select NRF_802154_FEM_PRESENT
+	help
+	  Enable FEM control and its configuration options.
+
+if IEEE802154_NRF5_FEM_PRESENT
+
+config IEEE802154_NRF5_FEM_PA_PIN_CTRL_ENABLE
+	bool "Enable PA pin control"
+	default y
+
+config IEEE802154_NRF5_FEM_PA_TIME_IN_ADVANCE_US
+	int "PA time advance"
+	default 13
+	help
+	  Time in microseconds when PA GPIO is activated before the radio
+	  is ready for transmission.
+
+config IEEE802154_NRF5_FEM_LNA_PIN_CTRL_ENABLE
+	bool "Enable LNA pin control"
+	default y
+
+config IEEE802154_NRF5_FEM_LNA_TIME_IN_ADVANCE_US
+	int "LNA time advance"
+	default 13
+	help
+	  Time in microseconds when LNA GPIO is activated before the radio
+	  is ready for reception.
+
+config IEEE802154_NRF5_FEM_PDN_PIN_CTRL_ENABLE
+	bool "Enable PDN pin control"
+	depends on NRF_802154_FEM_TYPE_3_PIN
+	default y
+
+config IEEE802154_NRF5_FEM_PDN_SETTLE_US
+	int "PDN settle time"
+	depends on NRF_802154_FEM_TYPE_3_PIN
+	default 18
+	help
+	  The time between activating the PDN and asserting RX_EN/TX_EN.
+
+config IEEE802154_NRF5_FEM_TRX_HOLD_US
+	int "TRX hold time"
+	default 5
+	help
+	  The time between deasserting the RX_EN/TX_EN and deactivating PDN.
+
+config IEEE802154_NRF5_FEM_PA_PIN
+	int "PA pin"
+	default 22
+
+config IEEE802154_NRF5_FEM_LNA_PIN
+	int "LNA pin"
+	default 19
+
+config IEEE802154_NRF5_FEM_PDN_PIN
+	int "PD pin"
+	default 23
+
+config IEEE802154_NRF5_FEM_MODE_PIN
+	int "Mode pin"
+	default 17
+
+config IEEE802154_NRF5_FEM_MOSI_PIN
```

Comment:
According to @tomchy this you need to set appropriate levels on SPI to let nRF21540 that you are using GPIO.

Reply:
Yes, that's what I found in the source file
This configuration is needed for PCA10112, because in this board all pins from nRF21540 are routed to nRF52. But on a custom board it is possible to route only needed pins to nRF52 and other connect directly to PCB ground / power supply to set to valid states. If SPI pins from nRF21540 are not connected to nRF52, we should not modify states of SPI pins in software.

Reply:
I understand now.